wrapanapi/systems/openstack.py

```diff
@@ -634,9 +634,23 @@ def tenant_api(self):
 
     @property
     def capi(self):
-        if not self._capi:
-            self._capi = cinderclient.Client(session=self.session, service_type="volume")
-        return self._capi
+        if self._capi:
+            return self._capi
+
+        # Detect the higest possible hopefully working cinder version
+        service_types = 'volumev3 volumev2 volume'.split()
+        for service_type in service_types:
+            try:
+                capi = cinderclient.Client(session=self.session, service_type=service_type)
+                capi.volumes.list()
+            except cinder_exceptions.EndpointNotFound as ex:
+                self.logger.debug('Cinder service type %s is not available: %s.', service_types, ex)
```

Comment:
I think you mean to use `service_type` here to log which version was not supported.